diff -urN desktop-common-data-2011.0/menu/applications.menu.in desktop-common-data-2011.0-patched/menu/applications.menu.in
--- desktop-common-data-2011.0/menu/applications.menu.in	2011-04-01 23:22:55.000000000 +1100
+++ desktop-common-data-2011.0-patched/menu/applications.menu.in	2011-08-18 01:58:16.714467891 +1100
@@ -35,6 +35,11 @@
   <Menu>
     <Name>Internet</Name>
     <Directory>mandriva-internet.directory</Directory>
+    <Layout>
+      <Merge type="menus"/>
+      <Merge type="files"/>
+      <Menuname>More</Menuname>
+    </Layout>
     <Include>
       <And>
         <Or>
@@ -229,6 +234,11 @@
   <Menu>
     <Name>SoundVideo</Name>
     <Directory>mandriva-soundvideo.directory</Directory>
+    <Layout>
+      <Merge type="menus"/>
+      <Merge type="files"/>
+      <Menuname>More</Menuname>
+    </Layout>
     <Include>
       <And>
         <Or>
@@ -301,10 +311,10 @@
       <Menuname inline="false">SystemTools</Menuname>
       <Menuname inline="false">Accessibility</Menuname>
       <Menuname inline="false">Emulators</Menuname>
+      <Merge type="menus"/>
       <Separator/>
       <Merge type="files"/>
       <Menuname>More</Menuname>
-      <Merge type="menus"/>
     </Layout>
     <Include>
       <And>
